src/rpc/starknet_backend.rs

```diff
@@ -0,0 +1,224 @@
+    fn get_class_hash_at(
+        &self,
+        block_id: BlockId,
+        contract_address: FeltParam,
+    ) -> RpcResult<Felt252> {
+        unimplemented!();
+    }
+
+    /// Implementation of the `syncing` RPC Endpoint.
+    async fn syncing(&self) -> RpcResult<SyncStatusType> {
+        unimplemented!();
+    }
+
+    /// Get the contract class definition in the given block associated with the given hash.
+    fn get_class(&self, block_id: BlockId, class_hash: FeltParam) -> RpcResult<ContractClass> {
+        unimplemented!();
+    }
+
+    /// Returns the specified block with transaction hashes.
+    fn get_block_with_tx_hashes(
+        &self,
+        block_id: BlockId,
+    ) -> RpcResult<MaybePendingBlockWithTxHashes> {
+        unimplemented!();
+    }
+
+    /// Get the nonce associated with the given address at the given block
+    fn get_nonce(&self, block_id: BlockId, contract_address: FeltParam) -> RpcResult<Felt252> {
+        unimplemented!();
+    }
+
+    /// Get block information with full transactions given the block id
+    fn get_block_with_txs(&self, block_id: BlockId) -> RpcResult<MaybePendingBlockWithTxs> {
+        unimplemented!();
+    }
+
+    /// Returns the chain id.
+    fn chain_id(&self) -> RpcResult<Felt252> {
+        unimplemented!();
+    }
+
+    /// Estimate the fee associated with transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - starknet transaction request
+    /// * `block_id` - hash of the requested block, number (height), or tag
+    ///
+    /// # Returns
+    ///
+    /// * `fee_estimate` - fee estimate in gwei
+    async fn estimate_fee(
+        &self,
+        request: Vec<BroadcastedTransaction>,
+        block_id: BlockId,
+    ) -> RpcResult<Vec<FeeEstimate>> {
+        unimplemented!();
+    }
+
+    // Returns the details of a transaction by a given block id and index
+    fn get_transaction_by_block_id_and_index(
+        &self,
+        block_id: BlockId,
+        index: u64,
+    ) -> RpcResult<Transaction> {
+        unimplemented!();
+    }
+
+    /// Get the information about the result of executing the requested block
+    fn get_state_update(&self, block_id: BlockId) -> RpcResult<StateUpdate> {
+        unimplemented!();
+    }
+
+    /// Returns all events matching the given filter
+    async fn get_events(&self, filter: EventFilterWithPage) -> RpcResult<EventsPage> {
+        unimplemented!();
+    }
+/// Add an Invoke Transaction to invoke a contract function
+    ///
+    /// # Arguments
+    ///
+    /// * `invoke tx` - <https://docs.starknet.io/documentation/architecture_and_concepts/Blocks/transactions/#invoke_transaction>
+    ///
+    /// # Returns
+    ///
+    /// * `transaction_hash` - transaction hash corresponding to the invocation
+    async fn add_invoke_transaction(
+        &self,
+        invoke_transaction: BroadcastedInvokeTransaction,
+    ) -> RpcResult<InvokeTransactionResult> {
+        unimplemented!();
```

Comment:
@igaray @juanbono we would want to insert transactions into the mempool here (and in `add_deploy_account_transaction` and `add_declare_transaction`)